security/monitoring.adoc

```diff
@@ -0,0 +1,196 @@
+[[security-monitoring]]
+= Monitoring Cluster Events and Logs
+{product-author}
+{product-version]
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[security-monitoring-intro]]
+== Introduction
+
+In addition to security measures mentioned in other sections of this
+guide, the ability to monitor and audit an {product-title} cluster is an
+important part of safeguarding the cluster and its users against
+inappropriate usage.
+
+There are two main sources of cluster-level information that
+are useful for this purpose: events and logs.
+
+[[security-monitoring-events]]
+== Cluster Events
+
+Cluster administrators are encouraged to familiarize themselves with the _Event_ resource
+type and review a xref:../dev_guide/events.adoc#events-reference[list of events] to
+determine which events are of interest. Depending on the master controller and plugin
+configuration, there are typically more potential event types than listed here.
+
+Events are associated with a namespace, either the namespace of the
+resource they are related to or, for cluster events, the _default_
+namespace. The _default_ namespace holds relevant events for monitoring or auditing a cluster,
+such as _Node_ events and resource events related to infrastructure components.
+
+The master API and `oc` command do not provide parameters to scope a listing of events to only those
+related to nodes. A simple approach would be to use grep:
+
+[source, bash]
+----
+$ oc get event -n default | grep Node
+1h         20h         3         origin-node-1.example.local   Node      Normal    NodeHasDiskPressure   ...
+----
+
+A more flexible approach is to output the events in a form that other
+tools can process. For example, the following example uses the `jq`
+tool against JSON output to extract only _NodeHasDiskPressure_ events:
+
+[source, bash]
+----
+$ oc get events -n default -o json \
+  | jq '.items[] | select(.involvedObject.kind == "Node" and .reason == "NodeHasDiskPressure")'
+
+{
+  "apiVersion": "v1",
+  "count": 3,
+  "involvedObject": {
+    "kind": "Node",
+    "name": "origin-node-1.example.local",
+    "uid": "origin-node-1.example.local"
+  },
+  "kind": "Event",
+  "reason": "NodeHasDiskPressure",
+  ...
+}
+----
+
+Events related to resource creation, modification, or deletion can also be good candidates for detecting misuse of the cluster. The following query, for
+example, can be used to look for excessive pulling of images:
+
+[source, bash]
+----
+$ oc get events --all-namespaces -o json \
+  | jq '[.items[] | select(.involvedObject.kind == "Pod" and .reason == "Pulling")] | length'
+
+4
+----
+
+[NOTE]
+====
+When a namespace is deleted, its events are deleted as well. Events can also expire and are deleted to prevent filling up the *etcd* file. Events are
```

Comment:
etcd is not a file, it's a database. Prefer to say "to prevent filling etcd storage."